src/test/test-core-unit.c

```diff
@@ -0,0 +1,100 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "escape.h"
+#include "tests.h"
+#include "unit.h"
+
+static void test_unit_escape_setting_one(
+                const char *s,
+                const char *expected_exec,
+                const char *expected_c) {
+
+        _cleanup_free_ char *a = NULL, *b, *c,
+                *s_esc, *a_esc, *b_esc, *c_esc;
```

Comment:
i think to avoid unnecessary confision I'd NULL initialize all

Reply:
We have the rule that in tests we don't initialize when the var is unconditionally set below. I think it's fine to leave it be. This code is covered 100% by the CI, so if the variable was accessed unitialized, that'd get reported immediately.